### PR TITLE
Fixes for aie2xclbin on Windows

### PR DESCRIPTION
**Pull request by erieaton-amd**
* Fix runtime errors when using aie2xclbin
* Remove libc dependency

### CODE REVIEW

File: tools/aie2xclbin/XCLBinGen.cpp [hunk 1]
```diff
@@ -299,17 +299,13 @@ static LogicalResult generateCoreElfFiles(ModuleOp moduleOp,
         std::string targetLower = StringRef(TK.TargetArch).lower();
         SmallVector<std::string, 10> flags;
         flags.push_back("-O2");
-        std::string targetFlag = "--target=" + targetLower + "-none-elf";
+        std::string targetFlag = "--target=" + targetLower;
```
Comment (stephenneuendorffer) [inline tools/aie2xclbin/XCLBinGen.cpp:302]: Should still have the 'none-elf' added here or it won't pick up the clang-rt IIRC.

**Review by stephenneuendorffer — APPROVED** [review]

File: tools/aie2xclbin/XCLBinGen.cpp [hunk 2]
```diff
@@ -299,17 +299,13 @@ static LogicalResult generateCoreElfFiles(ModuleOp moduleOp,
...
         flags.push_back(targetFlag);
         flags.emplace_back(objFile);
         SmallString<64> meBasicPath(TK.InstallDir);
         sys::path::append(meBasicPath, "aie_runtime_lib", TK.TargetArch,
                           "me_basic.o");
         flags.emplace_back(meBasicPath);
-        SmallString<64> libcPath(TK.PeanoDir);
```
Comment (stephenneuendorffer) [inline tools/aie2xclbin/XCLBinGen.cpp:309]: perhaps ifdef windows would be appropriate here, so we can still have full functionality on Linux?

File: tools/aie2xclbin/XCLBinGen.cpp (hunk 1 above)
Comment (erieaton-amd) [reply tools/aie2xclbin/XCLBinGen.cpp:302]: It looks in the wrong place for the builtins if I leave this as is:
`ld.lld: error: cannot open C:\s\llvm-aie\install\lib\clang\17\lib\libclang_rt.builtins-aie2.a: no such file or directory
clang: error: ld.lld command failed with exit code 1 (use -v to see invocation)`
The correct location is: `C:\s\llvm-aie\install\lib\clang\17\lib\aie2\libclang_rt.builtins.a`

Comment (stephenneuendorffer) [reply tools/aie2xclbin/XCLBinGen.cpp:304]: The correct location should include the full triple "aie2-unknown-none-elf".  So, you need to have the full target triple as the target when you're building compiler-rt (and libc).  This path seems to indicate that that didn't happen...

Comment (erieaton-amd) [reply tools/aie2xclbin/XCLBinGen.cpp:304]: The directions I got from Konstantin don't use the full triple, iirc.

**Pull request merged**